packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -658,7 +682,11 @@ function validateClassAttr(
     // Consequently, hydration mismatches will occur if scoped CSS token classnames
     // are rendered during SSR. This needs to be accounted for when validating.
     if (!isNull(scopeToken)) {
-        vnodeClasses.add(scopeToken);
+        if (vnodeClasses === EMPTY_SET) {
+            vnodeClasses = new Set([scopeToken]);
+        } else {
+            (vnodeClasses as Set<string>).add(scopeToken);
+        }
```

Comment:
Not quite sure the best optimization for empty except the scoped token, so I just went with this.